flake.nix

```diff
@@ -0,0 +1,51 @@
+{
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixos-unstable";
+    rust-overlay.url = "github:oxalica/rust-overlay";
+    flake-parts.url = "github:hercules-ci/flake-parts";
+  };
+
+  outputs = inputs@{ flake-parts, ... }:
+    flake-parts.lib.mkFlake { inherit inputs; }
+      {
+        systems = [
+          "x86_64-linux"
+          "aarch64-linux"
+        ];
+
+        perSystem = { self', lib, system, pkgs, config, ... }: {
+          _module.args.pkgs = import inputs.nixpkgs {
+            inherit system;
+
+            overlays = with inputs; [
+              rust-overlay.overlays.default
+            ];
+          };
+
+          packages = rec {
+            default = cargo-nfpm;
+            cargo-nfpm = pkgs.callPackage (import ./nix/package.nix) { };
+          };
+
+          devShells =
+            let
+              rust-toolchain = pkgs.rust-bin.fromRustupToolchainFile ./rust-toolchain.toml;
```

Comment:
Unfortunately, as a I suspected it, having `rust-toolchain.toml` pointing to `stable` will make the tests on nightly fail. Any possibility to achieve the same dev shell without requiring a `rust-toolchain.toml`? I suppose we could rename it?